monai/_extensions/hausdorff/hausdorff_cuda.cu

```diff
@@ -0,0 +1,2610 @@
+    int32_t* resultListPointerIterNumb;
+
+};
+
+
+/*
+* Basically holding the arguments for main kernel in the FullBoolPrep
+*/
+#pragma once
+template <typename TFB>
+struct ForBoolKernelArgs {
+    //matadata struct
+    MetaDataGPU metaData;
+    // dimensions of data block
+    int dbXLength;
+    int dbYLength;
+    int dbZLength;
+    // gold standard and segmentation output array
+    torch::Tensor goldArr;
+    torch::Tensor segmArr;
+    TFB numberToLookFor;
+    int Nx;
+    int Ny;
+    int Nz;
+
+
+    // Frequent accesses to "a" and "b"; infrequent accesses to "x" and "y":
+    //cuda::annotated_ptr<int const, cuda::access_property::persisting> a_p{ a }, b_p{ b };
+    //cuda::annotated_ptr<int, cuda::access_property::streaming> x_s{ x }, y_s{ y };
+
+    int32_t* resultListPointerMeta;
+    int32_t* resultListPointerLocal;
+    int32_t* resultListPointerIterNumb;
+
+    torch::Tensor resultListPointerIterNumbTensor;
+
+
+
+
+    uint32_t* origArrsPointer;
+    uint32_t* mainArrAPointer;
+    uint32_t* mainArrBPointer;
+    uint32_t* metaDataArrPointer;
+
+    uint32_t* workQueuePointer;
+    unsigned int* minMaxes;
+
+
+    /*
+main array with all required data  organized in sections for each metadata block
+x-  is block dimx times block dimy
+now what occupies what positions
+##mainArrA
+(0) - x-1 : reducedGoldRef
+(x) - 2x-1 : reducedSegmRef
+##mainArrB
+() - 3x-1 : reducedGoldPrev
+(x) - 4x-1 : reducedSegmPrev
+
+##metaDataArr
+0: empty
+1 :fpCount
+2 :fnCount
+3 :fpCounter
+4 :fnCounter
+5 :fpOffset
+6 :fnOffset
+7 :isActiveGold
+8 :isFullGold
+9 :isActiveSegm
+10 :isFullSegm
+11 :isToBeActivatedGold
+12 :isToBeActivatedSegm
+//now linear indexes of the blocks in all sides - if there is no block in given direction it will equal UINT32_MAX
+13 : top
+14 : bottom
+15 : left
+16 : right
+17 : anterior
+18 : posterior
+19 : empty
+20 : empty
+
+
+###origArrs
+0-x : reducedGold
+(x+1) - 2x : reducedSegm
+
+*/
+//   uint32_t* mainArr;
+
+
+
+
+    float robustnessPercent = 1.0;// 0.95;
+
+};
+
+
+
+
+
+//just utility for unit testing - set some data bout points
+#pragma once
+extern "C"  struct forTestPointStruct {
+    int x;
+    int y;
+    int z;
+
+    bool isGold;
+    bool isGoldAndSegm;
+
+    int xMeta;
+    int yMeta;
+    int zMeta;
+
+
+
+    bool shouldBeInResAfterOneDil;
+    bool shouldBeInResAfterTwoDil;
+    bool isFoundAndDilatated = false;
+    bool isFoundInResult = false;
+
+    bool isFoundDilTop = false;
+    bool isFoundDilBottom = false;
+
+    bool isFoundDilAnterior = false;
+    bool isFoundDilPosterior = false;
+
+    bool isFoundDilLeft = false;
+    bool isFoundDilRight = false;
+
+};
+
+
+#pragma once
+extern "C" struct forTestMetaDataStruct {
+
+    int xMeta;
+    int yMeta;
+    int zMeta;
+
+    int requiredspaceInFpResultList;
+    int requiredspaceInFnResultList;
+
+    bool isToBeActiveAtStart;
+    bool isToBeActiveAfterOneIter;
+    bool isToBeActiveAfterTwoIter;
```

Comment:
Could this (and the other members of these structs) be simplified to an integer (int activationIter)?

Reply:
You are referring just to test struct, I will remove it from C++ entirely; thanks !